visual-js/visual-playwright/src/api.ts

```diff
@@ -318,12 +329,64 @@ ${e instanceof Error ? e.message : JSON.stringify(e)}
       diffingMethod,
     });
 
-    await this.api.createSnapshot({
+    const { diffs } = await this.api.createSnapshot({
       ...meta,
       testName,
       suiteName,
       uploadUuid: uploadId,
     });
+
+    if (!this.uploadedDiffIds[testId]) this.uploadedDiffIds[testId] = [];
+
+    this.uploadedDiffIds[testId].push(...diffs.nodes.map((diff) => diff.id));
+  }
+
+  public async visualResults({ testId }: { testId: string }) {
```

Comment:
Agreed. IMO I'd like to keep it like it is now since some of the behavior for grouping / ID checking is different than the other implementation, but I'll create a ticket to unify it for all the JS frameworks.